pkg/security/probe/dentry_resolver.go

```diff
@@ -404,7 +421,7 @@ func (dr *DentryResolver) ResolveFromMap(mountID uint32, inode uint64, pathID ui
 		depth++
 
 		cacheKey = PathKey{MountID: key.MountID, Inode: key.Inode}
-		cacheEntry = PathEntry{Parent: path.Parent, Name: ""}
+		cacheEntry = dr.getPathEntryFromPool(path.Parent, "")
```

Comment:
I think you need to `Put()` the entries generated by this `Get()` if we encounter an error during the resolution loop (as those entries won't be inserted in the cache).